pkg/tfbridge/schema.go

```diff
@@ -479,51 +478,40 @@ func (ctx *conversionContext) makeTerraformInput(
 			oldObject = old.ObjectValue()
 		}
 
-		var input map[string]interface{}
-		if shimutil.IsOfTypeMap(tfs) {
-			var tfsElem shim.Schema
-			if tfs != nil {
-				if s, ok := tfs.Elem().(shim.Schema); ok {
-					tfsElem = s
-				}
-			}
-			var psElem *SchemaInfo
-			if ps != nil {
-				psElem = ps.Elem
-			}
-			var err error
-			input, err = ctx.makeMapTerraformInputs(oldObject, v.ObjectValue(),
-				tfsElem, psElem)
-			if err != nil {
-				return nil, err
-			}
-		} else {
-			var tfflds shim.SchemaMap
-			if tfs != nil {
-				if res, isres := tfs.Elem().(shim.Resource); isres {
-					tfflds = res.Schema()
-				}
+		var tfflds shim.SchemaMap
+
+		// We cannot use [shimutil.CastToTypeObject] because we have machinery that constructs invalid
+		// resource objects, such as [elemSchemas].
+		if tfs != nil {
+			if r, ok := tfs.Elem().(shim.Resource); ok {
+				tfflds = r.Schema()
 			}
+		}
+
+		if tfflds != nil {
 			var psflds map[string]*SchemaInfo
 			if ps != nil {
 				psflds = ps.Fields
 			}
-			var err error
-			input, err = ctx.makeObjectTerraformInputs(oldObject, v.ObjectValue(),
-				tfflds, psflds, rawNames)
+			obj, err := ctx.makeObjectTerraformInputs(oldObject, v.ObjectValue(), tfflds, psflds)
 			if err != nil {
-				return nil, err
+				return obj, err
```

Comment:
Should this really be under `if err!=nil`?

Reply:
Fixed.